parsing/parchmint-parser.js

```diff
@@ -128,14 +205,65 @@ class ParchmintParser {
     /**
      * Parse a Coord object from the given JSON object.
      *
+     * @since 1.0.0
+     *
      * @param {object}  coordObj    A JSON object with field x and y.
      * @returns {Coord} The resulting Coord object.
      */
     static parseCoord(coordObj) {
         return new Coord(coordObj['x'], coordObj['y']);
     }
 
+    /**
+     * Parse Ports from the given JSON array.
+     *
+     * Checks uniqueness of labels within the given port array and sets the
+     * parser invalid if they are not.
+     *
+     * @since 1.0.0
+     *
+     * @see parsePort
+     *
+     * @param {Array}  portsArr    A JSON array with Port fields.
+     * @returns {Map<string, Array>} A map containing all ports that have been
+     * parsed. The key is the ID of the layer on which the Port exists. The
+     * value is an Array of Port objects that exist on that layer.
+     */
+    parsePorts(portsArr) {
+        let portMap = new Map();
+        let idSet = new Set();
+
+        portsArr.forEach(value => {
+            // Check label uniqueness
+            if (idSet.has(value['label'])) {
+                this.valid = false;
+                console.log('Parser: Duplicate labels (' + value.id + ') exist in the Port list.')
+            } else {
+                idSet.add(value['label']);
+            }
+
+            // Parse port
+            if (!portMap.has(value['layer'])) {
+                portMap.set(value['layer'], [ParchmintParser.parsePort(value)]);
+            } else {
+                portMap.get(value['layer']).push(ParchmintParser.parsePort(value));
+            }
```

Comment:
If there isn't a ```label``` key, then it should be caught by the schema. If the ```label``` is invalid, then we can let it go and it will be caught by Port's ```validate``` method.